jiant/modules/seq2seq_decoder.py

```diff
@@ -185,10 +191,15 @@ def forward(
         output_dict = {"logits": logits}
 
         if target_tokens:
+            targets = target_tokens["words"]
             target_mask = get_text_field_mask(target_tokens)
-            loss = self._get_loss(logits, targets, target_mask)
-            output_dict["loss"] = loss
             output_dict["target_mask"] = target_mask
+            relevant_logits = logits[:, : targets.shape[1] - 1, :].contiguous()
```

Comment:
what is going on here exactly? Shouldn't the targets be padded such that you don't need to use `targets.shape[1]` (the loss should ignore them if you pass in the pad value ) ?

Reply:
Right; but in this PR the model generates up to maximum length (since the gold target sequence is not used). So we need to cut logits to the size of targets. The result for accuracy (which is the only thing we compute on this for now) is the same, since it is wrong anyways if <EOS> is not the last generated symbol.

Reply:
sgtm